src/subnet_groups.js

```diff
@@ -135,10 +135,22 @@ function buildDAXSubnetGroup(numZones = 0, { name = 'DAXSubnetGroup' } = {}) {
  * @param {Number} numZones Number of availability zones
  * @return {Object}
  */
-function buildSubnetGroups(numZones = 0) {
+function buildSubnetGroups(numZones = 0, subnetGroups = []) {
   if (numZones < 2) {
     return {};
   }
+  const subnetGroupList = {
+    rds: buildRDSSubnetGroup,
+    redshift: buildRedshiftSubnetGroup,
+    elasticache: buildElastiCacheSubnetGroup,
+    dax: buildDAXSubnetGroup,
+  }
+  if (subnetGroups.length > 0) {
+    return subnetGroups.reduce(function(acc, service) {
+      let builtSubnetGroup = subnetGroupList[service.toLowerCase()](numZones);
```

Comment:
Can we add a check here to ensure service is a valid key in subnetGroupList before trying to call it?

Reply:
I added it in `src/index.js`.
I think this is unavoidable to input validation logic into `src/index.js`.
because I don't want to get dirty importing `serverless.classes.Error` object on `src/subnet_groups.js`